### PR TITLE
Fixed mapping cosmos converter to handle value nodes

### PR DESCRIPTION
**Pull request by kushagraThapar**
Fixed mapping cosmos converter to handle value nodes

### CODE REVIEW

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java [hunk 1]
```diff
@@ -96,6 +99,27 @@ public void write(Object source, JsonNode sink) {
         }
     }
 
+    private <R> R readJsonNodeAsValue(JsonNode jsonNode, Class<R> type) {
+        switch (jsonNode.getNodeType()) {
+            case BOOLEAN:
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java:104]: Shouldn't we special case ARRAY here as well?

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java [hunk 2]
```diff
@@ -96,6 +99,27 @@ public void write(Object source, JsonNode sink) {
         }
     }
 
+    private <R> R readJsonNodeAsValue(JsonNode jsonNode, Class<R> type) {
```
Comment (mbhaskar) [inline sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java:102]: We can get rid of this code and just use objectMapper.treeToValue

**Review by mbhaskar — APPROVED** [review]
Please make the suggested code change 
Also please change the title to something like "Fixing cosmos converter to handle value nodes"

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java (hunk 2 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java]: Done.

File: sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java]: @FabianMeiswinkel - updated the logic to convert value nodes, objectMapper should automatically take care of that. Will test it out as well to verify.

**Pull request merged**